DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_priority.py

```diff
@@ -0,0 +1,102 @@
+    dut_ports = tgen_dev.links_dict[device][1]
+    vlan_priority = 7
+    vlan_id = 45
+
+    port_map = ({"port": 0, "settings": [{"vlan": vlan_id, "untagged": True, "pvid": True}]},
+                {"port": 1, "settings": [{"vlan": vlan_id, "untagged": True, "pvid": True}]},
+                {"port": 2, "settings": [{"vlan": vlan_id, "untagged": True, "pvid": True}]},
+                {"port": 3, "settings": [{"vlan": vlan_id, "untagged": True, "pvid": True}]})
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2. Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3. Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"],
+                dev_groups[tg_ports[3]][0]["name"]]
+
+    # 4. Setup packet stream(s).
+    streams = {f"Traffic with VLAN priority": {
+        "type": "raw",
+        "protocol": "802.1Q",
+        "ip_source": tx_ports,
+        "ip_destination": rx_ports,
+        "srcMac": "aa:bb:cc:dd:ee:ff",
+        "dstMac": "ff:ff:ff:ff:ff:ff",
+        "vlanID": vlan_id,
+        "vlanPriority": vlan_priority,
+        "egress_track_by": ["vlan_4k", "pcp"],
+    }}
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(5)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # 6. Verify traffic.
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+    for row in stats.Rows:
+        egress_stats = await tgen_utils_get_egress_stats(tgen_dev, row)
+        if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+            assert tgen_utils_get_loss(row) == 0.000, \
+                f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
+        else:
+            assert tgen_utils_get_loss(row) == 100.000, \
+                f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+        # 7. Verify priority was not stripped from the packet.
+        assert int(egress_stats.Rows[0]["VLAN:VLAN Priority"]) == vlan_priority, "No/wrong VLAN priority in the packet"
+        assert int(egress_stats.Rows[0]["VLAN:VLAN-ID"]) == vlan_id, "Wrong VLAN ID"
```

Comment:
you have to iterate over your egress stats rows:
```python
for e_row in egress_stats.Rows:
    ...
```
egress_stats.Rows[0] will just show you your regular stats. You have to check that egress_stats.Rows 1,2,3,etc... have the correct Rx value and check that there are no unexpected vlan/pcp values.

Reply:
If i'm getting `egress_stats.Rows[0]` i expect it to be egress already, no?

Reply:
If you look into what egress_stats is row-by-row you would see that `egress_stats.Rows[0]` is the same as the current `row`, but with additional `Egress Tracking` fields. And the other `egress_stats.Rows` contain the number of packets that matched a certain vlan and/or pcp value.